backend/src/users/views.ts

```diff
@@ -75,6 +75,47 @@ userRouter.post(
   }
 );
 
+userRouter.post(
+  "/googleCreate",
+  NoAuth as RequestHandler,
+  async (req: Request, res: Response) => {
+    // #swagger.tags = ['Users']
+    socketNotify("/users");
+    let user;
+    const { ...rest } = req.body;
+    try {
+      // Create user in local database
+      user = await userController.createUser(
+        rest,
+        rest.profile,
+        rest.preferences,
+        rest.permissions
+      );
+
+      // If local user doesn't exist, throw error
+      if (!user) {
+        throw Error("Failed to create local user");
+      }
+
+      // If test environment, return user without firebase auth
+      else if (process.env.NODE_ENV === "test") {
+        return res.status(201).send({ success: true, data: user });
+      }
+      
+    } catch (e: any) {
+      // If user exists in local database but not in Firebase, delete user
+      if (user) {
+        try {
+          await userController.deleteUser(user.id);
+        } catch (e: any) {
+          return res.status(500).send({ success: false, error: e.message });
+        }
+      }
+      return res.status(500).send({ success: false, error: e.message });
+    }
+  }
+);
+
```

Comment:
You don't need all this. Since it's just one action, simply use the patterns of `attempt` that other endpoints use

Reply:
You will actually need to set the user's custom claims in this controller